### ISSUE TITLE
ClassCastException in TerrainPatch.reIndexGeometry()

### ISSUE THREAD

**Issue by stephengold** [jMonkeyEngine/jmonkeyengine#1347]
JME 3.3.0-stable

```text
Apr 13, 2020 12:46:47 PM com.jme3.terrain.geomipmap.TerrainLodControl updateQuadLODs
SEVERE: null
java.util.concurrent.ExecutionException: java.lang.ClassCastException: java.nio.DirectByteBuffer cannot be cast to java.nio.ShortBuffer
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at com.jme3.terrain.geomipmap.TerrainLodControl.updateQuadLODs(TerrainLodControl.java:362)
	at com.jme3.terrain.geomipmap.TerrainLodControl.updateLOD(TerrainLodControl.java:281)
	at com.jme3.terrain.geomipmap.TerrainLodControl.controlUpdate(TerrainLodControl.java:221)
	at com.jme3.scene.control.AbstractControl.update(AbstractControl.java:111)
	at com.jme3.scene.Spatial.runControlUpdate(Spatial.java:737)
	at com.jme3.scene.Spatial.updateLogicalState(Spatial.java:880)
	at com.jme3.scene.Node.updateLogicalState(Node.java:231)
	at com.jme3.scene.Node.updateLogicalState(Node.java:242)
	at com.jme3.app.SimpleApplication.update(SimpleApplication.java:261)
	at com.jme3.system.lwjgl.LwjglWindow.runLoop(LwjglWindow.java:530)
	at com.jme3.system.lwjgl.LwjglWindow.run(LwjglWindow.java:632)
	at com.jme3.system.lwjgl.LwjglWindow.create(LwjglWindow.java:466)
	at com.jme3.app.LegacyApplication.start(LegacyApplication.java:463)
	at com.jme3.app.LegacyApplication.start(LegacyApplication.java:424)
	at com.jme3.app.SimpleApplication.start(SimpleApplication.java:127)
	at bats.BatsV2.mainStartup(BatsV2.java:402)
	at bats.BatsV2.main(BatsV2.java:186)
Caused by: java.lang.ClassCastException: java.nio.DirectByteBuffer cannot be cast to java.nio.ShortBuffer
	at com.jme3.terrain.geomipmap.TerrainPatch.reIndexGeometry(TerrainPatch.java:260)
	at com.jme3.terrain.geomipmap.TerrainQuad.reIndexPages(TerrainQuad.java:535)
	at com.jme3.terrain.geomipmap.TerrainQuad.reIndexPages(TerrainQuad.java:533)
	at com.jme3.terrain.geomipmap.TerrainQuad.reIndexPages(TerrainQuad.java:533)
	at com.jme3.terrain.geomipmap.TerrainQuad.reIndexPages(TerrainQuad.java:533)
	at com.jme3.terrain.geomipmap.TerrainLodControl$UpdateLOD.call(TerrainLodControl.java:484)
	at com.jme3.terrain.geomipmap.TerrainLodControl$UpdateLOD.call(TerrainLodControl.java:455)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by stephengold** [comment]
Nobody spoke up, so I'll fix this myself.

**Comment by stephengold** [comment]
In master branch, this particular crash was fixed as part of PR #1285. However, code search revealed 2 more places in master-branch jme3-terrain where `ByteBuffer` wasn't supported, so I'll fix those, then start cherry-picking to the v3.3 branch.

**Comment by stephengold** [comment]
The other 2 places were fixed in master branch at a6f2c127 .

**Comment by stephengold** [comment]
This particular crash fixed in v3.3 branch at 98a283b .  The other 2 places were fixed in v3.3 branch at 3a5f966 .
